### PR TITLE
Clean refactor pre-2fa

### PR DESCRIPTION
**Pull request by mattlockyer**
Moves some methods required by 2fa out of app.js and into
- utils/sms.js
- utils/email.js
- utils/near.js

In the second commit, these functions are removes from app.js and imported instead.

The is a require refactor pre-2fa since 2fa depends on these utils as well.

### CODE REVIEW

**Comment by mattlockyer** [comment]
Looks like checks have passed. Is there anything else blocking?

This is the minimum refactor of shared methods needed to be broken out before I can add more 2fa PRs.

File: utils/near.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+const nearPromise = (async () => {
+    const near = await nearAPI.connect({
+        deps: { keyStore },
+        masterAccount: creatorKeyJson && creatorKeyJson.account_id,
+        nodeUrl: process.env.NODE_URL
+    });
+    return near;
+})();
+
+const getContract = async (contractName, viewMethods, changeMethods, secretKey) => {
```
Comment (vgrichina) [inline utils/near.js:27]: where is `getContract` called?

File: app.js [hunk 2]
```diff
@@ -35,28 +34,11 @@ app.use(async function(ctx, next) {
...
-        return null;
-    }
-})();
-const keyStore = {
-    async getKey() {
-        return nearAPI.KeyPair.fromString(creatorKeyJson.secret_key || creatorKeyJson.private_key);
-    }
-};
+const {
+    creatorKeyJson,
```
Comment (vgrichina) [inline app.js:38]: Makes sense to extract all the code using `creatorKeyJson` into module and just export middleware handling account creation. Any reason not to do it?

File: app.js [hunk 3]
```diff
@@ -35,28 +34,11 @@ app.use(async function(ctx, next) {
...
+    nearPromise,
```
Comment (vgrichina) [inline app.js:39]: `nearPromise` also used only in one middleware, makes sense to export middleware `withNear` instead

File: utils/near.js (hunk 1 above)
Comment (mattlockyer) [reply utils/near.js:27]: In an upcoming PR. For now it's just a helper method, shouldn't really be an issue to include it here.

File: app.js (hunk 2 above)
Comment (mattlockyer) [reply app.js:38]: Seems like a majority of the near code should go in `middleware/near.js` vs. `utils/near.js` correct?

And we can tighten this: https://github.com/near/near-contract-helper/blob/d70186d6f37625eedf238e048c2a2042e143745f/app.js#L37
Into a proper middleware `withNear` as you mentioned in: https://github.com/near/near-contract-helper/pull/165/files/d70186d6f37625eedf238e048c2a2042e143745f#r445233017

File: app.js (hunk 3 above)
Comment (mattlockyer) [reply app.js:39]: Answered in https://github.com/near/near-contract-helper/pull/165/files/d70186d6f37625eedf238e048c2a2042e143745f#r445239440

File: utils/near.js (hunk 1 above)
Comment (vgrichina) [reply utils/near.js:27]: Let's include it when it's actually used. E.g. it probably shouldn't use `creatorKeyJson` at all. 

Same actually might apply to `withNear` as well. Looks like we ideally should depend on `creatorKeyJson` only in middleware which is using it to create accounts.

File: utils/near.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        async getKey() {
+            return nearAPI.KeyPair.fromString(secretKey);
+        },
+    };
+    const near = await nearAPI.connect({
+        deps: { keyStore },
+        masterAccount: creatorKeyJson && creatorKeyJson.account_id,
+        nodeUrl: process.env.NODE_URL
+    });
+    const contractAccount = new nearAPI.Account(near.connection, contractName);
```
Comment (vgrichina) [inline utils/near.js:38]:
```js
const contractAccount = await near.account(contractName);
```

File: app.js (hunk 2 above)
Comment (vgrichina) [reply app.js:38]: yeah, seems good to me

**Comment by mattlockyer** [comment]
Closing to trigger a new build on Render with a new PR for the same branch.

**Comment by mattlockyer** [comment]
New PR here to trigger Render build: https://github.com/near/near-contract-helper/pull/188

**Pull request closed without merging**